cloudflare/resource_cloudflare_teams_route.go

```diff
@@ -0,0 +1,134 @@
+package cloudflare
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareTeamsRoute() *schema.Resource {
+	return &schema.Resource{
+		Schema: resourceCloudflareTeamsRouteSchema(),
+		Create: resourceCloudflareTeamsRouteCreate,
+		Read:   resourceCloudflareTeamsRouteRead,
+		Update: resourceCloudflareTeamsRouteUpdate,
+		Delete: resourceCloudflareTeamsRouteDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareTeamsRouteImport,
+		},
+	}
+}
+
+func resourceCloudflareTeamsRouteRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	tunnelRoute, err := client.GetTunnelRouteForIP(context.Background(), cloudflare.TunnelRoutesForIPParams{
+		AccountID: d.Get("account_id").(string),
+		Network:   d.Get("network").(string),
+	})
+	if err != nil {
+		// FIXME(2022-04-21): Until the API returns a valid v4 compatible envelope, we need to
+		// check if the error message is related to problems unmarshalling the response _or_
+		// an expected not found error.
+		var notFoundError *cloudflare.NotFoundError
+		if strings.Contains(err.Error(), "error unmarshalling the JSON response error body") || errors.As(err, &notFoundError) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("error reading Tunnel Route for Network %q: %w", d.Id(), err)
+	}
+
+	d.Set("tunnel_id", tunnelRoute.TunnelID)
+	d.Set("network", tunnelRoute.Network)
+
+	if len(tunnelRoute.Comment) > 0 {
+		d.Set("comment", tunnelRoute.Comment)
+	}
+
+	return nil
+}
+
+func resourceCloudflareTeamsRouteCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	resource := cloudflare.TunnelRoutesCreateParams{
+		AccountID: d.Get("account_id").(string),
+		TunnelID:  d.Get("tunnel_id").(string),
+		Network:   d.Get("network").(string),
+	}
+
+	if comment, ok := d.Get("comment").(string); ok {
+		resource.Comment = comment
+	}
+
+	newTunnelRoute, err := client.CreateTunnelRoute(context.Background(), resource)
+	if err != nil {
+		return fmt.Errorf("error creating Tunnel Route for Network %q: %w", d.Get("network").(string), err)
+	}
+
+	d.SetId(newTunnelRoute.Network)
+
+	return resourceCloudflareTeamsRouteRead(d, meta)
+}
+
+func resourceCloudflareTeamsRouteUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	resource := cloudflare.TunnelRoutesUpdateParams{
+		AccountID: d.Get("account_id").(string),
+		TunnelID:  d.Get("tunnel_id").(string),
+		Network:   d.Get("network").(string),
+	}
+
+	if comment, ok := d.Get("comment").(string); ok {
```

Comment:
If you have a comment in a resource and then simply delete the line with the intent to remove the comment, will it remove it?  Perhaps the comment parameter should have a default of `""` so it's always set to something since the API seems to always return an empty string as the comment field.